lib/gitlab/ci/templates/pipeline/AndroidTemplate.yml

```diff
@@ -140,9 +155,32 @@ assembleApkRelease:
   artifacts:
     paths:
       - $DEPLOY_RELEASE_APK_PATH
+      - $DEPLOY_RELEASE_AAR_PATH
+
+deployArtifactorySnapshot:
+  stage: deploy
+  rules: *snapshot_rules
+  script:
+    - ./gradlew tasks | grep publish -q
+    - if [ $? -eq 0 ]; then
+        ./gradlew publish -PartifactoryUsername=$CTI_USERNAME -PartifactoryPassword=$CTI_PASSWORD -PartifactoryURL="${ARTIFACTORY_SNAPSHOT_URL}"
+      else
+        echo "No publish command found";
+      fi
+
+deployArtifactoryRelease:
+  stage: deploy
+  rules: *release_rules
+  script:
+    - ./gradlew tasks | grep publish -q
+    - if [ $? -eq 0 ]; then
+        ./gradlew publish -PartifactoryUsername=$CTI_USERNAME -PartifactoryPassword=$CTI_PASSWORD -PartifactoryURL="${ARTIFACTORY_RELEASE_URL}"
+      else
+        echo "No publish command found";
+      fi
```

Comment:
Have you tested out that a missing publish task doesn't cause the pipeline job to fail?